lib/ex_doc/formatter/html/templates/head_template.eex

```diff
@@ -26,8 +26,15 @@
     <script>
       <%# Immediately apply night mode preference to avoid a flash effect %>
       try {
-        if (localStorage.getItem('night-mode') === 'true') {
-          document.body.classList.add('night-mode');
-        }
+        var settings = localStorage.getItem('ex_doc:settings');
+        settings = JSON.parse(settings || '{}');
+        var isDark = null;
+
+        if (localStorage.getItem('night-mode') === 'true') { isDark = true }
+        if (settings.nightMode === true) { isDark = true }
+        if (settings.theme === 'dark') { isDark = true }
+        if (settings.theme === 'system' && window.matchMedia('(prefers-color-scheme: dark)').matches) { isDark = true }
+        if (isDark === null && settings.theme === null && window.matchMedia('(prefers-color-scheme: dark)').matches) { isDark = true }
+        if (isDark) { document.body.classList.add('dark') }
```

Comment:
Yeah, in my opinion it is fine to remove the legacy setting. It may even be preferable because most people have configured their OS to dark mode and we should just pick their preferences if the legacy stuff is removed. So can you please remove any legacy lookup? Then this is good to go. :) Thank you!

Reply:
updated